lib/internal/streams/writable.js

```diff
@@ -337,19 +338,20 @@ function WritableState(options, stream, isDuplex) {
 }
 
 function resetBuffer(state) {
-  state.buffered = [];
+  state.buffered = null;
```

Comment:
As mentioned a better approach would be to use the fact every object is also an array and just store on state[0] state[1] etc and instead keep a length property. This should be smaller than maintaining an actual array.

Reply:
Not sure I understand

Reply:
`buffer = { length: 0, 0: "", 1: "" };` will allocate less memory than `buffer = ["", ""]` and the Array prototype is not used in our case so it will work the same

Reply:
That will make things complicated give doWrite.